lib/CMakeLists.txt

```diff
@@ -201,44 +212,76 @@ target_compile_definitions(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,COMPILE_DEFIN
 target_include_directories(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,INCLUDE_DIRECTORIES>)
 target_compile_options(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,COMPILE_OPTIONS>)
 
-# add target specific files		
-if(${QUDA_TARGET_TYPE} STREQUAL "CUDA")		
-  add_subdirectory(targets/cuda)		
-endif()		
-if(${QUDA_TARGET_TYPE} STREQUAL "HIP")		
-  #add_subdirectory(targets/hip)
-endif()		
+# add target specific files
+if(${QUDA_TARGET_TYPE} STREQUAL "CUDA")
+  add_subdirectory(targets/cuda)
+endif()
+if(${QUDA_TARGET_TYPE} STREQUAL "HIP")
+  # add_subdirectory(targets/hip)
+endif()
 target_sources(quda PRIVATE $<TARGET_OBJECTS:quda_target>)
- 
+
 add_subdirectory(targets/generic)
 target_sources(quda PRIVATE $<TARGET_OBJECTS:quda_generic>)
- 
+
 # propagate CXX flags to CUDA host compiler
+#TODO: Do we still need that? 
 if(${QUDA_PROPAGATE_CXX_FLAGS})
-  target_compile_options(
-    quda
-    PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:
-           $<$<CONFIG:DEVEL>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_DEVEL}>
-           $<$<CONFIG:STRICT>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_STRICT}>
-           $<$<CONFIG:RELEASE>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_RELEASE}>
-           $<$<CONFIG:DEBUG>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_DEBUG}>
-           $<$<CONFIG:HOSTDEBUG>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_HOSTDEBUG}>
-           $<$<CONFIG:DEVICEDEBUG>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_DEVICEDEBUG}>
-           >)
+
+  # Pick the right set of flags Apparently I cannot do this with generator expressions
+  if(CMAKE_BUILD_TYPE STREQUAL "DEVEL")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_DEVEL}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "STRICT")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_STRICT}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_RELEASE}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_DEBUG}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "HOSTDEBUG")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_HOSTDEBUG}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "DEVICEDEBUG")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_DEVICEDEBUG}")
+  endif()
+
+  # Turne the flags into a CMAKE list
+  string(REPLACE " " ";" PROPAGATED_FLAG_LIST ${PROPAGATED_FLAGS})
```

Comment:
This line throws an error with `SANITIZE` builds. Issue seems to be `SANITIZE` isn't included in the above set of `if` commands. 

Specific error:
```
CMake Error at lib/CMakeLists.txt:247 (string):
  string sub-command REPLACE requires at least four arguments.
```

Relevant part of `cmake` command is:
```
cmake [...] -DCMAKE_BUILD_TYPE=SANITIZE [...]
```

Reply:
Balint will fix, will wait to mark this as closed until the fix is in.

Reply:
I have added a clause to this clause, and pushed it but likely it in and of itself may not have broken the build as the only flag  on this at the moment would've been a "-g" as things stand and the -fsanitize flags get added later. It wouldn't give you a line number from the UB and ASAN traps tho.

Reply:
Let's wait for the builds to finish